server/game/cards/06-WoE/Befuddle.js

```diff
@@ -16,7 +16,15 @@ class Befuddle extends Card {
                 targetController: 'opponent',
                 effect: ability.effects.playerCannot(
                     'play',
-                    (sourceContext) => !sourceContext.source.hasHouse(context.house)
+                    (sourceContext) =>
+                        // When an in-play source plays another card, for
+                        // some reason the play restriction is still
+                        // checked on that card, even though it's already
+                        // in play.  So ignore those restriction checks
+                        // for cards already in play.
+                        sourceContext.source.location !== 'play area' &&
+                        // A card can't be played if it has any house that's not the called house.
+                        sourceContext.source.getHouses().some((h) => h !== context.house)
```

Comment:
It is subtle and took me a while to see, but they don't do the same thing.  Imagine a card with two houses: ['logos', 'staralliance'].  Then say the Befuddle player called house 'logos', meaning that any card that's not of house 'logos' cannot be played.

* `!card.hasHouse('logos')` will return `false` (due to the `!`), and so in the original code, the card will be able to be played.
* `card.getHouses().some((h) => h !== context.house)` will return `true`, because there is one house that is not logos.  So the card will _not_ be able to be played.

I'm not sure which code you mean is duplicate-y.  We could introduce a new `notHasHouse()` helper if you prefer.

Reply:
I should clarify: the helper function would be called 'hasNotHouse()` or, more clearly, `hasSomeHouseThatIsNot()`.

Reply:
hasHouse() uses lots of the same code as getHouse(). You're new hasNoHouse() could also use the common code that getHouse uses to first get the list of houses then work out what it's interested in

Reply:
Ok, updated with a new helper function for this logic.  I didn't really follow your point about using common code; we can just call the whole `getHouses()` function here without factoring out any additional helper functions.

Reply:
Followed up in discord